src/plugins/intel_gpu/src/kernel_selector/cl_kernels/gather_nonzero_ref.cl

```diff
@@ -9,18 +9,27 @@
 #define VSTORE CAT(vstore,VSIZE)
 #define OUTPUT_VTYPE MAKE_VECTOR_TYPE(OUTPUT_TYPE, VSIZE)
 
-KERNEL (gather_nonzero_ref)(const __global INPUT0_TYPE* input,
-                            volatile __global INPUT1_TYPE* output_shape,
-                            __global OUTPUT_TYPE* output)
+KERNEL (gather_nonzero_ref)(
+    OPTIONAL_SHAPE_INFO_ARG
+    const __global INPUT0_TYPE* input,
+    volatile __global INPUT1_TYPE* output_shape,
+    __global OUTPUT_TYPE* output)
 {
     int local_offset = 0;
     const int result_size = OV_INPUT_RANK * OUTPUT_FEATURE_NUM; // output shape: [ov_rank, count_nonzero]
 
-#ifdef USE_LOCAL_MEM
+#if IS_DYNAMIC
+    const int dst_slm_size = TOTAL_DATA_SIZE * OV_INPUT_RANK;
+    #define DST_SLM_SIZE dst_slm_size
+#if DST_SLM_SIZE < MAX_LOCAL_MEM_SIZE
```

Comment:
This is compile time preprocessor, not be able to be judged at runtime. I guess that your attention is to check this condition at runtime. For that purpose, USE_LOCAL_MEM should be handled as a variable, not the macro.

Reply:
updated as your suggestion.